plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/error_resolution/CustomMarkerResolutionGenerator.java

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Primetals Technologies Austria GmbH
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Sebastian Hollersbacher
+ *    - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package org.eclipse.fordiac.ide.application.error_resolution;
+
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.fordiac.ide.model.errormarker.FordiacErrorMarker;
+import org.eclipse.fordiac.ide.model.libraryElement.ErrorMarkerFBNElement;
+import org.eclipse.ui.IMarkerResolution;
+import org.eclipse.ui.IMarkerResolutionGenerator2;
+
+public class CustomMarkerResolutionGenerator implements IMarkerResolutionGenerator2 {
+
+	@Override
+	public IMarkerResolution[] getResolutions(final IMarker marker) {
+		try {
+			final EObject obj = FordiacErrorMarker.getTarget(marker);
+			if (obj instanceof final ErrorMarkerFBNElement errorType) {
+				return new IMarkerResolution[] { new CreateMissingFBMarkerResolution(errorType),
+						new ChangeFBMarkerResolution(errorType) };
+			}
```

Comment:
The `FordiacErrorMarker.getTarget(marker)` will load the file if necessary, which can be a considerable draw of resources and impact overall performance.

I would suggest to rely on the target type attribute instead, pass the marker to the resolutions and use `FordiacErrorMarker.getTarget(marker)` only when actually fixing the marker (or looking for other markers).
```suggestion
			final EClass targetType = FordiacErrorMarker.getTargetType(marker);
			if (LibraryElementPackage.Literals.ERROR_MARKER_FBN_ELEMENT.equals(targetType)) {
				return new IMarkerResolution[] { new CreateMissingFBMarkerResolution(marker),
						new ChangeFBMarkerResolution(marker) };
			}
```